test/join.jl

```diff
@@ -719,4 +719,128 @@ end
+        mu in [true, false], vl in [true, false], vr in [true, false]
+        @test rightjoin(df1, df2, on=[:id1, :id2 => :ID2],
+                       makeunique = mu, validate = vl => vr, rename = l => r) ≅
+            DataFrame(id1=[1,2,4], id2=[1,2,4], x_left=[1,2,missing], x_right=[1,2,3])
+    end
+
+    @test_throws ArgumentError rightjoin(df1, df2, on=[:id1, :id2 => :ID2],
+                                         rename = "_left" => "_right", indicator=:id1)
+    @test_throws ArgumentError rightjoin(df1, df2, on=[:id1, :id2 => :ID2],
+                                         rename = "_left" => "_right", indicator=:x_left)
+    @test rightjoin(df1, df2, on=[:id1, :id2 => :ID2],
+                    rename = "_left" => "_right", indicator=:ind) ≅
+          DataFrame(id1=[1,2,4], id2=[1,2,4], x_left=[1,2,missing],
+                    x_right=[1,2,3], ind=["both", "both", "right_only"])
+
+    @test_throws ArgumentError outerjoin(df1, df2, on=:id1)
+    @test outerjoin(df1, df2, on=:id1, makeunique=true) ≅
+        DataFrame(id1=[1,2,3,4], id2=[1,2,3,missing], x=[1,2,3,missing],
+                  ID2=[1,2,missing,4], x_1=[1,2,missing,3])
+    for l in ["_left", :_left, x -> x * "_left"],
+        r in ["_right", :_right, x -> x * "_right"],
+        mu in [true, false], vl in [true, false], vr in [true, false]
+        @test outerjoin(df1, df2, on=:id1,
+                       makeunique = mu, validate = vl => vr, rename = l => r) ≅
+            DataFrame(id1=[1,2,3,4], id2_left=[1,2,3,missing], x_left=[1,2,3,missing],
+                      ID2_right=[1,2,missing,4], x_right=[1,2,missing,3])
+    end
+
+    @test_throws ArgumentError outerjoin(df1, df2, on=[:id1, :id2 => :ID2])
+    @test outerjoin(df1, df2, on=[:id1, :id2 => :ID2], makeunique=true) ≅
+        DataFrame(id1=[1,2,3,4], id2=[1,2,3,4], x=[1,2,3,missing], x_1=[1,2,missing,3])
+    for l in ["_left", :_left, x -> x * "_left"],
+        r in ["_right", :_right, x -> x * "_right"],
+        mu in [true, false], vl in [true, false], vr in [true, false]
+        @test outerjoin(df1, df2, on=[:id1, :id2 => :ID2],
+                       makeunique = mu, validate = vl => vr, rename = l => r) ≅
+            DataFrame(id1=[1,2,3,4], id2=[1,2,3,4], x_left=[1,2,3,missing], x_right=[1,2,missing,3])
+    end
+
+    @test_throws ArgumentError outerjoin(df1, df2, on=[:id1, :id2 => :ID2],
+                                         rename = "_left" => "_right", indicator=:id1)
+    @test_throws ArgumentError outerjoin(df1, df2, on=[:id1, :id2 => :ID2],
+                                         rename = "_left" => "_right", indicator=:x_left)
+    @test outerjoin(df1, df2, on=[:id1, :id2 => :ID2],
+                    rename = "_left" => "_right", indicator=:ind) ≅
+          DataFrame(id1=[1,2,3,4], id2=[1,2,3,4], x_left=[1,2,3,missing],
+                    x_right=[1,2,missing,3], ind=["both", "both", "left_only", "right_only"])
+end
```

Comment:
Can you test what happens if a column with the target name already exists, and when the two renaming functions return duplicate values (either duplicating its own names, or duplicating the other function's names)? This also includes passing twice the same suffix.

Reply:
I have added tests with and without `makeunique`.